client/src/main/java/com/tigrisdata/db/client/StandardTigrisCollection.java

```diff
@@ -128,26 +126,18 @@ public Optional<T> readOne(TransactionSession session, TigrisFilter filter)
   }
 
   @Override
-  public Iterator<SearchResult<T>> search(SearchRequest request, SearchRequestOptions options)
+  public Optional<SearchResult<T>> search(SearchRequest request, SearchRequestOptions options)
       throws TigrisException {
-    Api.SearchRequest apiSearchRequest =
-        TypeConverter.toSearchRequest(databaseName, collectionName, request, options, objectMapper);
-    try {
-      Iterator<Api.SearchResponse> resp = blockingStub.search(apiSearchRequest);
-      Function<SearchResponse, SearchResult<T>> converter =
-          r -> SearchResult.from(r, objectMapper, collectionTypeClass);
-      return Utilities.transformIterator(resp, converter);
-    } catch (StatusRuntimeException statusRuntimeException) {
-      throw new TigrisException(
-          SEARCH_FAILED,
-          TypeConverter.extractTigrisError(statusRuntimeException),
-          statusRuntimeException);
+    Iterator<SearchResult<T>> resultIterator = this.searchInternal(request, options);
```

Comment:
this iterator will always be of one item?

Reply:
[Yes](https://github.com/tigrisdata/tigris/blob/main/server/services/v1/query_runner.go#L570-L573)

Reply:
you don't have to do it now, but please cover this in test. This API change should trigger update in test/test failure.